hnix-store-remote/src/System/Nix/Store/Remote.hs

```diff
@@ -32,3 +46,23 @@ verifyStore :: CheckFlag -> RepairFlag -> MonadStore Bool
 verifyStore check repair = simpleOpArgs VerifyStore $ do
   putBool check
   putBool repair
+
+addToStore :: forall hashAlgo. (NamedAlgo hashAlgo, ValidAlgo hashAlgo)
+    => StorePathName
+    -> FilePath
+    -> RecursiveFlag
+    -> PathFilter
+    -> RepairFlag
+    -> MonadStore LBS.ByteString
```

Comment:
A single `store` can only be associated with a single prefix, so IMO `MonadStore` should have its prefix associated at the type level (and we can return a proper path here)

Reply:
From what I can see, there is no way to find the storeDir that a daemon uses. This seems to be the reason for all the assertStorePath in remote-store.cc. In a sense this configuration defeats the purpose of having the storeDir prefix in the Store type, as we will have to chack everything that the daemon returns for compliance with our own storeDir prefix.

I will make an attempt and wait for your feedback :-)

Reply:
Seems to work, even if type annotations are a bit cumbersome. See tests/Operations.hs.